pkg/controller/controller.go

```diff
@@ -61,9 +60,9 @@ type Controller struct {
 	handler              *Handler
 	serviceManager       ServiceManager
 	configRefreshChan    chan string
-	provider             base.Provider
+	provider             *provider.Provider
```

Comment:
Can we keep this an interface so that we can ensure that an interface can be used by EE for integration?